src/docs/Dockerfile

```diff
@@ -0,0 +1,51 @@
+## Base ########################################################################
+# Use a larger node image to do the build for native deps (e.g., gcc, python)
+FROM node:lts as base
+
+# Reduce npm log spam and colour during install within Docker
+ENV NPM_CONFIG_LOGLEVEL=warn
+ENV NPM_CONFIG_COLOR=false
+
+# We'll run the app as the `node` user, so put it in their home directory
+WORKDIR /home/node/app
+# Copy the package.json and lock file over
+COPY package*.json /home/node/app/
+
+## Development #################################################################
+# Define a development target that installs devDeps and runs in dev mode
+FROM base as development
+WORKDIR /home/node/app
+# Install (not ci) with dependencies, and for Linux vs. Linux Musl (which we use for -alpine)
+RUN npm install
+# Copy the source code over
+COPY --chown=node:node . /home/node/app/
+# Switch to the node user vs. root
+USER node
+# Expose port 4631
+EXPOSE 4631
+# Start the app in debug mode so we can attach the debugger
+CMD ["npm", "run", "start"]
+
+## Production ##################################################################
+# Also define a production target which doesn't use devDeps
+FROM base as production
+WORKDIR /home/node/app
+# We'll install production only deps
+RUN npm install --production
+
+## Deploy ######################################################################
+# Use a stable nginx image
+FROM nginx:stable as deploy
+# https://github.com/krallin/tini
+RUN apk --no-cache add tini
```

Comment:
I'm having problems building the docker container through `pnpm run services:start`.

I am not very familiar with `docker`, but I think these are the problematic parts. The `deploy` image we are using uses `debian` instead of `alpine`, which makes the line `RUN apk ...` to fail: `apk` does not exist. I tried to change it to `apt-get install tini`, but the package cannot be found (despite the `tini` GitHub repo saying otherwise...).

Maybe we could use an `alpine` image when building everything, and then place the `nginx` image right before the `COPY`ing step, just like we do in `src/web/Dockerfile`?

Reply:
We don't need `tini` at all, we can get rid of this completely, since we aren't actually running a process.  `tini` is for running the node process in a container.  In this case, we build the site with node, then copy the results into the nginx container's filesystem where it expects to find it.

We could change `nginx:stable` to `nginx:stable-alpine` though, to save some size.

Reply:
Thank you for the reviews. I'm not a Docker pro so I copied and pasted some of @humphd work from #2808 . Also removed tini and changed `nginx:stable` to `nginx:stable-alpine`